app/src/main/java/org/groundplatform/android/MainActivity.kt

```diff
@@ -83,7 +89,19 @@ class MainActivity : AbstractActivity() {
 
     viewModel = viewModelFactory[this, MainViewModel::class.java]
 
-    lifecycleScope.launch { viewModel.navigationRequests.filterNotNull().collect { updateUi(it) } }
+    lifecycleScope.launch {
+      viewModel.navigationRequests.filterNotNull().first()
```

Comment:
What's the purpose of this line?

Reply:
to make sure we have the data before we act on the intent data, to understand the current navigation request ongoing.

Reply:
Won't this block the UI when opening the app without a deeplink?

Reply:
Yes, we did test, with roberto also, this is not blocking the UI.